src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -652,78 +656,57 @@ class npc_dark_rider_of_acherus : public CreatureScript
+                events.Update(diff);
 
-                if (PhaseTimer <= diff)
+                switch (uint32 eventId = events.ExecuteEvent())
                 {
-                    switch (Phase)
-                    {
-                       case 0:
-                            Talk(SAY_DARK_RIDER);
-                            PhaseTimer = 5000;
-                            Phase = 1;
-                            break;
-                        case 1:
-                            if (Unit* target = ObjectAccessor::GetUnit(*me, TargetGUID))
-                                DoCast(target, SPELL_DESPAWN_HORSE, true);
-                            PhaseTimer = 3000;
-                            Phase = 2;
-                            break;
-                        case 2:
-                            me->SetVisible(false);
-                            PhaseTimer = 2000;
-                            Phase = 3;
-                            break;
-                        case 3:
-                            me->DespawnOrUnsummon();
-                            break;
-                        default:
-                            break;
-                    }
+                    case EVENT_START_MOVE:
+                        me->SetTarget(HorseGUID);
+                        me->SetSpeedRate(MOVE_RUN, 0.4f);
+
+                        if (Creature* horse = ObjectAccessor::GetCreature(*me, HorseGUID))
+                            me->GetMotionMaster()->MoveFollow(horse, 2.0f, 0.f);
+                        break;
+                    case EVENT_DESPAWN_HORSE:
+                        Talk(SAY_DARK_RIDER);
+                        if (Creature* horse = ObjectAccessor::GetCreature(*me, HorseGUID))
+                            DoCast(horse, SPELL_DESPAWN_HORSE, true);
+                        break;
+                    case EVENT_END_SCRIPT:
+                        me->DespawnOrUnsummon();
+                        break;
+                    default:
+                        break;
                 }
-                else
-                    PhaseTimer -= diff;
             }
-
-            void InitDespawnHorse(Unit* who)
+			
+			void SpellHitTarget(Unit* target, SpellInfo const* spell) override
             {
-                if (!who)
-                    return;
-
-                TargetGUID = who->GetGUID();
-                me->SetWalk(true);
-                me->SetSpeedRate(MOVE_RUN, 0.4f);
-                me->GetMotionMaster()->MoveChase(who);
-                me->SetTarget(TargetGUID);
-                Intro = true;
+                if (spell->Id == SPELL_DESPAWN_HORSE && target->GetGUID() == HorseGUID)
+                    if (Creature* creature = target->ToCreature())
+                        creature->DespawnOrUnsummon(2500);
             }
 
         private:
-            uint32 PhaseTimer;
-            uint32 Phase;
-            bool Intro;
-            ObjectGuid TargetGUID;
+            ObjectGuid HorseGUID;
```

Comment:
_horseGUID;
_events;

private members need has _ prefix

Reply:
Ok, just renamed the previous ones, without the underscore :)